deploy/ansible/roles/process_bom/tasks/iterate_bom.yml

```
@@ -1,38 +1,45 @@
 ---
 
-- name: "Ensure {{ base_bom }} is used"
+- name: "Ensure {{ bom_base_name }} is available"
+  uri:
+    url: "{{ sapbits_location_base_path }}/boms/{{ bom_base_name }}/bom.yml"
+    dest: "/tmp/{{ bom_base_name }}_bom.yml"
+    creates: "/tmp/{{ bom_base_name }}_bom.yml"
+  delegate_to: localhost
+
+- name: "Ensure {{ bom_base_name }} is used"
   include_vars:
-    file:       "{{ base_bom }}"
-    name:       bom
+    file: "/tmp/{{ bom_base_name }}_bom.yml"
+    name: bom
+  delegate_to: localhost
+
+- name: "Ensure {{ target_media_location }} exists"
+  file:
+    path: "{{ target_media_location }}"
+    state: directory
 
-- name: Group iterable tasks together
+- name: "Ensure current property is iterated"
   vars:
-    bom_dependencies: "{{ bom.materials.dependencies is defined | ternary(bom.materials.dependencies, []) }}"
-    archive_location: "{{ bom.defaults.archive_location | regex_replace('/$', '') }}"
-    target_location: "{{ bom.defaults.target_location | regex_replace('/$', '') }}"
-  block:
-    - name: Ensure media property is iterated
-      vars:
-        file_source: "{{ archive_location }}/{{ media_iterator.archive }}"
-        file_dest_target_location: "{{ media_iterator.override_target_location is defined | ternary(media_iterator.override_target_location, target_location) | regex_replace('/$', '') }}"
-        file_dest_target_name: "{{ media_iterator.override_target_filename is defined | ternary(media_iterator.override_target_filename, media_iterator.archive) }}"
-        file_dest: "{{ file_dest_target_location }}/{{ file_dest_target_name }}"
-        bom_media: '{{ bom.materials.media }}'
-      # uri:
-        # url: "{{ file_source }}"
-        # dest: "{{ file_dest }}"
-        # creates: "{{ file_dest }}"
-      debug:
-        msg: "Download {{ file_source }} to {{ file_dest }}"
-      loop: "{{ bom_media | flatten(levels=1) }}"
-      loop_control:
-        loop_var: media_iterator
+    default_location: "{{ bom.defaults.target_location | regex_replace('/$', '') }}"
+    bom_media: "{{ media_iterator.media }}"
+    source_location: "{{ media_iterator.location }}"
+    file_name: "{{ media_iterator.property_name }}"
+  include_tasks: download_media.yml
+  loop:
+    - location: archives
+      property_name: archive
+      media: "{{ bom.materials.media | flatten(levels=1) }}"
+    - location: templates
+      property_name: file
+      media: "{{ bom.materials.templates | flatten(levels=1) }}"
```

To add in the stackfile processing:

```suggestion
      media: "{{ bom.materials.templates | flatten(levels=1) }}"
    - location: "boms/{{ bom_base_name }}/stackfiles"
      property_name: file
      media: "{{ bom.materials.stackfiles | flatten(levels=1) }}"

```

---

Note: This requires the `stackfiles` block of the bom to be defined.  I had to add it in to my HANA bom.

---

My PR #58 adds the stackfiles into the example BoMs. I was going to update the script to add them. I've also got this as an unpushed change.